eth-contracts/contracts/CKBChain.sol

```diff
@@ -69,6 +73,33 @@ contract CKBChain is ICKBChain, ICKBSpv {
 
     /// #ICKBSpv
     function proveTxExist(bytes calldata txProofData, uint64 numConfirmations) external view returns (bool){
+        bytes29 proofView = txProofData.ref(uint40(ViewSpv.SpvTypes.CKBTxProof));
+        uint64 blockNumber = proofView.spvBlockNumber();
+        bytes32 blockHash = proofView.blockHash();
+        require(blockNumber + numConfirmations <= lastBlockNumber, "blockNumber is too ahead");
+        require(canonicalHeaderHashes[blockNumber] == blockHash, "blockNumber and blockHash mismatch");
+        require(canonicalTransactionRoots[blockHash] != bytes32(0), "blockHash invalid or too old");
+        uint16 index = proofView.txMerkleIndex();
+        uint16 sibling;
+        uint256 lemmasIndex = 0;
+        bytes29 lemmas = proofView.lemmas();
+        uint256 length = lemmas.len() / 32;
+        bytes32 res = proofView.txHash();
+
+        while (lemmasIndex < length && index > 0) {
+            sibling = ((index + 1) ^ 1) - 1;
+            if (index < sibling) {
+                res = CKBCrypto.digest(abi.encodePacked(res, lemmas.indexH256Array(lemmasIndex), new bytes(64)), 64);
+            } else {
+                res = CKBCrypto.digest(abi.encodePacked(lemmas.indexH256Array(lemmasIndex), res, new bytes(64)), 64);
+            }
+
+            lemmasIndex++;
+            // index = parent(index)
+            index = (index - 1) >> 1;
+        }
+        res = CKBCrypto.digest(abi.encodePacked(res, proofView.witnessesRoot(), new bytes(64)), 64);
```

Comment:
What is the `new bytes(64)` for?

Reply:
The blake2b implementation requires that the data length must be an integer multiple of 128. I will optimize this Blake2b implementation after CKBChain contracts finished

Reply:
Then add a todo and some comments for the `digest` function.